pennylane/spin/lattice.py

```diff
@@ -190,7 +200,61 @@ def _generate_grid(self, neighbour_order):
             lattice_points.append(point)
             lattice_map.append(node_index)
 
-        return math.array(lattice_points), math.array(lattice_map)
+        return math.array(lattice_points), lattice_map
+
+    def get_custom_edges(self, custom_edges, lattice_map):
+        """Generates the edges described in `custom_edges` for all unit cells."""
+
+        if not all([len(edge) in (1, 2) for edge in custom_edges]):
+            raise TypeError(
+                """
+                custom_edges must be a list of tuples of length 1 or 2.
+                Every tuple must contain two lattice indices to represent the edge
+                and can optionally include a list to represent the operation and coefficient for that edge.
```

Comment:
A little confused. So custom_edges is supposed to be a list of tuples `[(), ()...]`, where the tuples are of length 1 or 2, and they should contain two lattice indices and optionally include a list.

So if the tuple length can only be 1 or 2, but this reads like length can be 3 e.g.`[(lattice_index, lattice_index, [operation, coefficient])]`? I'm confused.

An example would help.

Reply:
no the first part is a tuple containing the edge information as in the vertices, the second is the tuple containing operation and coefficient, the second tuple is optional.